### ISSUE TITLE
Continue department on the next page if it didn't fit

### ISSUE THREAD

**Issue by mvasin** [jakubboucek/schedule-reception-doctors#5]
Sometimes rows are half-shown, like on this screenshot.

<img width="1434" alt="screen shot 2017-03-06 at 14 33 54" src="https://cloud.githubusercontent.com/assets/12434833/23608348/04ce942e-027a-11e7-87df-7f29e1989cc1.png">

I guess it's closely related to another issue: we should fix the department that didn't fit on the previous page; it should continue on the next page, repeating the table header.

**Comment by mvasin** [comment]
Departments' names are cut as well:

<img width="1293" alt="screen shot 2017-03-06 at 15 19 34" src="https://cloud.githubusercontent.com/assets/12434833/23609725/72cc6f7c-0280-11e7-8a54-cc5d8a3f8f79.png">

**Comment by jakubboucek** [comment]
Please, what is this size dimensions of screen? 

Paging is separated to monitor when is page reload. When you change resolution, you MUST  reload page.

**Comment by mvasin** [comment]
I do refresh the page after screen resizing.

At the moment, if I make 1096 × 494 viewport (I check it here http://viewportsizes.com/mine/), then refresh the page, the very first page of the schedule looks like this:

<img width="1064" alt="screen shot 2017-03-06 at 16 13 19" src="https://cloud.githubusercontent.com/assets/12434833/23611353/e42c5ac2-0287-11e7-9b6b-7732a1710ad8.png">

**Comment by jakubboucek** [comment]
Uf oh, that is very small screen :-( I expected minimal resolution like an 1200x768, separate of tables row is major change. Currently is the table minimal atomic item.

Can you temporary make content page smallest?
![How to make page smallest](https://cloud.githubusercontent.com/assets/1657322/23613045/10c95d6e-027e-11e7-8edd-6ddbf78c32ff.png)

Ad: "Departments' names are cut“ - that's bug, I'll fix that.

**Comment by mvasin** [comment]
We can expect the TV screen to be no less than "HD ready" 1280×720 resolution, but what if the number of doctors per department will be really big? That's my major concern.

**Comment by jakubboucek** [comment]
We did not specify the precise conditions here :) 

OK, my expect was wrong, i will thing how to make it better. Unreservedly: I am afraid to crumble table to more pages because it can mean more trouble and unxepexted states.

Is good enough for you when big table will slow scroll content whole available viewport?

**Comment by mvasin** [comment]
Scrolling the too long table maybe a little weird, as doesn't match the normal behaviour of the rest of our app (just changing the pages).

But if splitting tables is really so troublesome, and bearing in mind the one-table-overflowing-the-whole-screen situation will not happen too often, maybe never for most clinics, I'd say - OK.

**Comment by mvasin** [comment]
Maybe monitoring if page has scrolled till the end before switching to another page is more troublesome then splitting tables?

**Comment by jakubboucek** [comment]
It is easier because it is manupulating only with one DOM node and `animate()` method know callback when is done.

Now I decrease priority of this issue for move with another. This case is good case for my research  at long dreamless nights. :)

**Comment by 7flash** [comment]
What's the problem now?

**Comment by mvasin** [comment]
@7flash The problem is: if a department's table is too long, it will not fit the screen, and will be cut.

So we need either:

-  to split the table (it would be better), repeating the table header on the next page,
- or to make the table scroll slowly from top to bottom (it's worse, because it's not consistent with the rest of the interface, that is just fading in - fading out pages one by one).

**Comment by 7flash** [comment]
I done this already
 - to split the table (it would be better), repeating the table header on the next page,

**Comment by mvasin** [comment]
It's not in the current codebase, please make pull request if you have it.

**Comment by 7flash** [comment]
I will try

**Comment by 7flash** [comment]
I created a new branch with smart pagination but with manual setting variable resourcesPerPage. It's without smart fit to screen size. Do you have ideas how to merge it?

**Comment by jakubboucek** [comment]
Preview of branch is for acception deployed to: https://s3-eu-west-1.amazonaws.com/schedule-reception-doctors/smart-pagination/index.html

**Comment by jakubboucek** [comment]
Hi @7flash, hi reviewed your commit 392135ed455be8b2a0a01eff389235f9544bf4a6:

What does mean `resourcesPerPage`? If is it the [same thing as in origin code](https://github.com/jakubboucek/schedule-reception-doctors/blob/e05ccfb1609cb0b8b79e906a798efa7ad4c562fb/source/app.js#L71), that IMHO doe's not sense. Each resource have different (and unpredictable) height and must be counted by content's and viewport's height.

**Comment by mvasin** [comment]
@7flash Did you just paste your old code, overwriting your own #8 PR and ignoring all the work done by @jakubboucek? Man, that's stupid.

Don't ignore the current code structure with `app.js` file.
Improve current 'smart pagination' (adapting to screen size), don't drag us back to 'dumb  pagination' (setting number of rows per page manually).

**Comment by 7flash** [comment]
Setting number of rows per page manually is a good choice. Each resource could have equally size and resourcesPerPage counts titles. 
I see no other options to implement it. Do you have ideas?

**Comment by mvasin** [comment]
@7flash No, setting rows manually is a bad choice. Some rows will be two (or more) rows tall, some not, it's never predictable.

It's already implemented by @jakubboucek, now we need to make it legacy browsers compatible.
